File: My_AndroidPagingLibrary/app/src/main/java/com/android/hootr/my_androidpaginglibrary/MainActivity.java
package com.android.hootr.my_androidpaginglibrary;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.paging.PagedList;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.android.hootr.my_androidpaginglibrary.model.User;
import com.android.hootr.my_androidpaginglibrary.ui.UserAdapter;
import com.android.hootr.my_androidpaginglibrary.viewmodel.UserViewModel;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        final UserAdapter adapter = new UserAdapter();
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        UserViewModel itemViewModel = ViewModelProviders.of(this).get(UserViewModel.class);
        itemViewModel.userPagedList.observe(this, new Observer<PagedList<User>>() {
            @Override
            public void onChanged(PagedList<User> users) {
                adapter.submitList(users);
            }
        });
        recyclerView.setAdapter(adapter);
    }

}

File: MyTestRoomNewArrival/app/src/main/java/com/android/hootr/mytestroomnewarrival/MainActivity.kt
package com.android.hootr.mytestroomnewarrival

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var viewModel: TestViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewModel = ViewModelProvider(this).get(TestViewModel::class.java)

        viewModel.listNewArrival.observe(this, Observer {
//            Log.i("happy", it.toString())
        })

        viewModel.listFullArrival.observe(this, Observer {
            Log.i("happy", it.toString())
        })

        btnTest.setOnClickListener {
            viewModel.testInsertData()
        }

    }
}
